packages/cli/src/commands/doctor/healthchecks/reactNativeGlobal.ts

```diff
@@ -26,42 +30,48 @@ const findGlobalPackage = (packageManager, moduleId) => {
   }
 };
 
-const checkGlobalPaths = moduleId => {
-  return !!(
-    (findGlobalPackage(packageManagers.yarn, moduleId) ||
-      findGlobalPackage(packageManagers.npm, moduleId)) &&
-    true
-  );
+const checkGlobalInstallalations = () => {
+  //Returns an array of globally installed paths for rn and rncli both in yarn and npm
+  const librariesGloballyInstalled = Object.keys(packages)
+    .map(packageKey =>
+      Object.keys(packageManagers).map(packageManagerKey =>
+        findGlobalPackage(
+          packageManagers[packageManagerKey],
+          packages[packageKey],
+        ),
+      ),
+    )
+    .filter(e => e != null);
+
+  //Cant's use flat() in ts...? This is an alternative
+  const flattenedArray = [].concat.apply([], librariesGloballyInstalled);
+
+  console.log('!!!!!! ' + flattenedArray);
+
+  return flattenedArray;
 };
 
-const checkGlobalInstall = () => {
-  let reactNativeCLIGlobal;
-  let reactNativeGlobal;
+const removeNodePackage = async (packagePath, loader) => {
+  console.log('******removing ' + packagePath);
 
-  try {
-    reactNativeGlobal = checkGlobalPaths('react-native');
-  } catch (error) {
-    return null;
-  }
-
-  try {
-    reactNativeCLIGlobal = checkGlobalPaths('react-native-cli');
-    logger.log('******Found r-n-cli: ' + reactNativeCLIGlobal);
-  } catch (error) {
-    return null;
-  }
+  const packageManager = packagePath.includes('yarn') ? 'yarn' : 'npm';
+  const packageName = packagePath.includes('react-native/')
```

Comment:
Instead of doing this I suggest that you return the `packageName` in the loop on line 35.

Reply:
Sorry not sure I understand how this will work, are you suggesting that librariesGloballyInstalled (line 29) returns an array of packageNames only that are globally installed, and as such checkGlobalInstallations returns these? 

As far as I'm aware, we need to know both which packages are globally installed and with which package managers? Returning only the packageName would mean that we lose the package manager information, but both of these are contained in the current return value which is the path of the globally installed package? Sorry if I have misunderstood